subprojects/core/src/main/java/org/gradle/api/internal/initialization/loadercache/DeprecatingClassLoader.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.initialization.loadercache;
+
+import org.gradle.internal.classloader.ImplementationHashAware;
+import org.gradle.internal.classpath.ClassPath;
+import org.gradle.internal.hash.HashCode;
+import org.gradle.util.DeprecationLogger;
+
+import java.net.URL;
+
+public class DeprecatingClassLoader extends ClassLoader implements ImplementationHashAware {
+
+    private final ClassLoader deprecatedUsageLoader;
+    private final ClassLoader nonDeprecatedParent;
+
+    public DeprecatingClassLoader(ClassPath classPath, ClassLoader deprecatedUsageLoader, ClassLoader nonDeprecatedParent) {
+        super(null);
+        this.deprecatedUsageLoader = deprecatedUsageLoader;
+        this.nonDeprecatedParent = nonDeprecatedParent;
+    }
+
+    @Override
+    public Class<?> loadClass(String name) throws ClassNotFoundException {
+        return loadClass(name, false);
+    }
+
+    @Override
+    public URL getResource(String name) {
+        URL resource = nonDeprecatedParent.getResource(name);
+        if (resource != null) {
+            return resource;
+        }
+        resource = deprecatedUsageLoader.getResource(name);
+        if (resource != null) {
+            DeprecationLogger.nagUserOfDeprecated("Using buildSrc resources in settings", "Do not use '" + name + "' in settings.");
+        }
+        return resource;
+    }
+
+    @Override
+    protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+        try {
+            return nonDeprecatedParent.loadClass(name);
+        } catch (ClassNotFoundException e) {
+            // Expected
+        }
+
+        try {
+            Class<?> deprecatedUsageClass = deprecatedUsageLoader.loadClass(name);
+            DeprecationLogger.nagUserOfDeprecated("Using buildSrc classes in settings", "Do not use '" + name + "' in settings.");
```

Comment:
WDYT about how "spammy" this might be?

Reply:
quite spammy I think. I did this initial deprecation warning very detailed in order to make it easier to test in the spike. We probably just want a more generic deprecation warning. That would also allow to use that generic warning about using _something_ from buildSrc in settings when a plugin is applied and avoid that _weired_ pointer to a class I havnt explicitly used in the settings as I only know the plugin id

Reply:
changed it to be one same warning without the class name as details. this way its way less spammy and you only get it once